explainaboard/processors/named_entity_recognition.py

```diff
@@ -351,12 +328,33 @@ def _complete_features(
         :return: The features that are active (e.g. skipping training set features when
             no training set available)
         """
+        sys_features = unwrap(sys_info.features)
+        active_features = list(
+            sys_features.get_bucket_features(
+                include_training_dependent=external_stats is not None
+            )
+        )
+
+        sent_feats: list[str] = []
+        tok_feats: list[str] = []
+        for x in active_features:
+            (sent_feats if (x in sys_features) else tok_feats).append(x)
+
         for _id, dict_sysout in tqdm(enumerate(sys_output), desc="featurizing"):
             # Get values of bucketing features
             tokens = dict_sysout["tokens"]
 
             # sentence_length
             dict_sysout["sentence_length"] = len(tokens)
+            self._get_max_min_value(sys_info, "sentence_length", len(tokens))
+
+            # entity density
+            dict_sysout["entity_density"] = len(
+                BIOSpanOps().get_spans(tags=dict_sysout["true_tags"])
```

Comment:
`BIOSpanOps` can probably be initialized outside of the for loop?

Reply:
good capture